src/pages/core/external-plugins/app-data.md

```diff
@@ -248,3 +349,178 @@ let write_to_app_data_plugin_ix = WriteExternalPluginAdapterDataV1CpiBuilder::ne
 {% /dialect %}
 
 {% /dialect-switcher %}
+
+## Reading Data from the AppData Plugin
+
+Data can be both read on chain progams and external sources pulling account data.
+
+### Fetch the Raw Data
+
+The first step to deserializing the data stored in an AppData plugin is to fetch the raw data and check the schema field which dictates the format in which the data is stored before serialization.
+
+{% dialect-switcher title="Fetching AppData Raw Data" %}
+{% dialect title="JavaScript" id="js" %}
+
+```ts
+const assetId = publicKey('11111111111111111111111111111111')
+const dataAuthority = publicKey('33333333333333333333333333333333')
+
+const asset = await fetchAsset(umi, assetId)
```

Comment:
So for Rust client we recommend they only deserialize the parts they need.  But for JS don't we give them everything?  Doesn't `fetchAsset` deserialize the asset and all the internal and external plugins for them, plus parse the `AppData's` data?

I think the flow is `fetchAssetV1` -> `deserializeAssetV1` -> `getAssetV1AccountDataSerializer()` ->  `externalRegistryRecordsToExternalPluginAdapterList` -> `appDataFromBase` -> `parseExternalPluginAdapterData`.

So for typescript wouldn't we show just using fetchAsset and what they would get as a result which I think would be the data too?  Or am I missing something?

Reply:
Full JS example for reference.

```ts
const assetId = publicKey('11111111111111111111111111111111')
const dataAuthority = publicKey('33333333333333333333333333333333')

const asset = await fetchAsset(umi, assetId)

let appDataPlugin = asset.appDatas?.filter(
  (appData) => (appData.authority.address = dataAuthority)
)

let data
let schema

// Check if `AppData` plugin with the given authority exists
if (appDataPlugin && appDataPlugin.length > 0) {
  // Save plugin data to `data`
  data = appDataPlugin[0].data

  // Save plugin schema to `schema`
  schema = appDataPlugin[0].schema
}
```

> So for JS since we typically don't have the same memory constraints as a Rust program user, wouldn't we show just using fetchAsset and what they would get as a result which I think would be the data too? Or am I missing something?

`FetchAsset()` does return a fully deserialized asset but this example shows you how to grab a specific `AppData` plugin on the asset like we do in the Rust version. Maybe not needed?

Reply:
Yeah I believe its already deserialized, so then maybe you don't need to show that in JS because you already have it, unless its binary and you wanted to show using something extra like bincode.